Added Reference Property attribute

armmarov opened the pull request:
1. The modification allows user to set their own external references. 
2. If the external reference does not exist, it will use the internal set reference template.

armmarov committed:
Add Reference Property feature

armmarov committed:
Update README.md

armmarov committed:
Update package.json

armmarov committed:
Update objectcleaner.html

Stwissel commented:
Commit has a few issues:
- When specifying a reference property, template could be empty, but leaving it empty creates an error
- Help not updated - how should user know what reference property is?

fix and I'll merge

armmarov committed:
Add function for sanity checking, Reindent using prettier jslint

armmarov commented:
Hi Stephan, I already added a commit to doing a sanity check on both external reference property and internal template. This will allow any of this to be empty, but at the same time preventing both fields to be empty.
Added the help information regarding the reference property, please let me know if it is still insufficient.

FYI, I use prettier jslint to reindent the file.
